client/service-container-chain/src/rpc.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) Moondance Labs Ltd.
+// This file is part of Tanssi.
+
+// Tanssi is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
+
+//! A collection of node-specific RPC methods.
+//! Substrate provides the `sc-rpc` crate, which defines the core RPC layer
+//! used by Substrate nodes. This file extends those RPC definitions with
+//! capabilities that are specific to this project's runtime configuration.
+
+#![warn(missing_docs)]
+
+pub use sc_rpc::DenyUnsafe;
+use {
+    cumulus_primitives_core::ParaId,
+    dancebox_runtime::{opaque::Block, AccountId, Index as Nonce},
+    manual_xcm_rpc::{ManualXcm, ManualXcmApiServer},
+    polkadot_primitives::Hash,
+    sc_client_api::{AuxStore, UsageProvider},
+    sc_consensus_manual_seal::{
+        rpc::{ManualSeal, ManualSealApiServer},
+        EngineCommand,
+    },
+    sc_transaction_pool_api::TransactionPool,
+    services_payment_rpc::{
+        ServicesPayment, ServicesPaymentApiServer as _, ServicesPaymentRuntimeApi,
+    },
+    sp_api::ProvideRuntimeApi,
+    sp_block_builder::BlockBuilder,
+    sp_blockchain::{Error as BlockChainError, HeaderBackend, HeaderMetadata},
+    std::sync::Arc,
+    stream_payment_rpc::{StreamPayment, StreamPaymentApiServer as _, StreamPaymentRuntimeApi},
+};
+
+/// A type representing all RPC extensions.
+pub type RpcExtension = jsonrpsee::RpcModule<()>;
+
+/// Full client dependencies
+pub struct FullDeps<C, P> {
+    /// The client instance to use.
+    pub client: Arc<C>,
+    /// Transaction pool instance.
+    pub pool: Arc<P>,
+    /// Whether to deny unsafe calls
+    pub deny_unsafe: DenyUnsafe,
+    /// Manual seal command sink
+    pub command_sink: Option<futures::channel::mpsc::Sender<EngineCommand<Hash>>>,
+    /// Channels for manual xcm messages (downward, hrmp)
+    pub xcm_senders: Option<(flume::Sender<Vec<u8>>, flume::Sender<(ParaId, Vec<u8>)>)>,
```

Comment:
@ParthDesai this might be of interest for you, specially for the starlight manual seal